SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -201,9 +201,9 @@ - (instancetype)initWithConfiguration:(SDLConfiguration *)configuration delegate
 }
 
 - (void)startWithReadyHandler:(SDLManagerReadyBlock)readyHandler {
-    dispatch_sync(_lifecycleQueue, ^{
+    [self sdl_runOnProcessingQueue:^{
```

Comment:
It turns out that I was not re-starting the exiting `SDLManager` object but rather creating a new `SDLManager` object. This creates a new lifecycle manager but the old lifecycle manager is still around. Both lifecycle managers get the `SDLRPCServiceDidConnect` notification but only one is in the `started` state, hence the crash. Perhaps it would be better to add the notifications in when the `SDLManager` is started and remove the notifications when the `SDLManager` is stopped?

Reply:
We don't currently support multiple `SDLManager`'s existing at the same time, and I think your "bug" is basically attempting to do that. You created a new manager and didn't deallocate the old one. I changed the lifecycle manager to listen only to its own notification dispatcher, but not everything in the manager will necessarily work.

I think your suggestion could potentially lead to other issues where notifications that should be received are not. We cannot just remove the notification handlers when the developer says `stop` because we'll miss notifications. We would have to add it into `sdl_stopManagersAndRestart:`, but only when the manager won't be reconnected. So that wouldn't cover your use-case if the old manager is restarted. I think that what you did is just unsupported.